polyfills/IntersectionObserver/polyfill.js

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2016 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// TODO: rootMargin are completely ignored for now
+(function(scope) {
+  if('IntersectionObserver' in window) {
+    return;
+  }
+
+  var POLL_INTERVAL = 100;
+
+  // Constructor
+  var IntersectionObserver = function(callback, options) {
+    options = options || {};
+
+    if(!(callback instanceof Function)) {
+      throw('callback needs to be a function');
+    }
+
+    if(options.root && !(options.root instanceof HTMLElement)) {
+      throw('Root needs to be an HTMLElement');
+    }
+
+    this._callback = callback;
+    this._root = options.root || null;
+    this._rootMargin = options.rootMargin || [0, 0, 0, 0];
+    this._thresholds = options.threshold || [0];
+    this._init();
+  };
+
+  IntersectionObserver.prototype = {
+    //
+    // Public API
+    //
+    get root() {
```

Comment:
I believe this is ES6/2015 method definition syntax. Do we need to change this to ES5?